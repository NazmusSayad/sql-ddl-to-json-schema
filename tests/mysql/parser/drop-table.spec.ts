import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should drop people table': {
    queries: [
      'DROP TABLE people;',
      'drop table people;',
      'drop temporary table people;',
      'drop temporary table if exists people;',
      'drop temporary table if exists people restrict;',
      'drop temporary table if exists people cascade;',
    ],
  },

  'Parser: Should drop several tables': {
    queries: [
      'DROP TABLE people, cars,pets ,test;',
      'drop table people, cars,pets ,test;',
      'drop temporary table people, cars,pets ,test;',
      'drop temporary table if exists people, cars,pets ,test;',
      'drop temporary table if exists people, cars,pets ,test restrict;',
      'drop temporary table if exists people, cars,pets ,test cascade;',
    ],
  },
});
