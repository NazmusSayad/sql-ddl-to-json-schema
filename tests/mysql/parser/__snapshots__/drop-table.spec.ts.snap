// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should drop people table 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop people table 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop people table 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop people table 4`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop people table 5`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop people table 6`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop several tables 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people",
          "cars",
          "pets",
          "test"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop several tables 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people",
          "cars",
          "pets",
          "test"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop several tables 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people",
          "cars",
          "pets",
          "test"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop several tables 4`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people",
          "cars",
          "pets",
          "test"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop several tables 5`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people",
          "cars",
          "pets",
          "test"
        ]
      }
    }
  ]
}
`;

exports[`Parser: Should drop several tables 6`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_DROP_TABLE",
        "def": [
          "people",
          "cars",
          "pets",
          "test"
        ]
      }
    }
  ]
}
`;
